kube-core/src/cel.rs

```diff
@@ -280,3 +282,137 @@ pub fn merge_properties(s: &mut Schema, merge: &mut Schema) {
         }
     }
 }
+
+/// ListType represents x-kubernetes merge strategy for list.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum ListMerge {
+    /// Atomic represents a list, where entire list is replaced during merge. At any point in time, a single manager owns the list.
+    Atomic,
+    /// Set applies to lists that include only scalar elements. These elements must be unique.
+    Set,
+    /// Map applies to lists of nested types only. The key values must be unique in the list.
+    Map(Vec<String>),
+}
+
+/// MapMerge represents x-kubernetes merge strategy for map.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum MapMerge {
+    /// Atomic represents a map, which can only be entirely replaced by a single manager.
+    Atomic,
+    /// Granular represents a map, which supports separate managers updating individual fields.
+    Granular,
+}
+
+/// StructMerge represents x-kubernetes merge strategy for struct.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum StructMerge {
+    /// Atomic represents a struct, which can only be entirely replaced by a single manager.
+    Atomic,
+    /// Granular represents a struct, which supports separate managers updating individual fields.
+    Granular,
+}
+
+/// MergeStrategy represents set of options for a server-side merge strategy applied to a field.
```

Comment:
```suggestion
/// MergeStrategy represents set of options for a server-side merge strategy applied to a field.
///
/// See upstream documentation of values at https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy
```

probably good to link upstream here